File: app-java/src/main/java/com/tritiumstudios/phasmophobiaevidencetool/android/views/account/AccountDetailsView.java
package com.tritiumstudios.phasmophobiaevidencetool.android.views.account;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;

import com.tritiumstudios.phasmophobiaevidencetool.R;
import com.tritiumstudios.phasmophobiaevidencetool.android.firebase.firestore.transactions.user.FirestoreUser;
import com.google.firebase.auth.FirebaseUser;

public class AccountDetailsView extends ConstraintLayout {

    public AccountDetailsView(@NonNull Context context) {
        super(context);

        init(getContext());
    }

    public AccountDetailsView(@NonNull Context context,
                              @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs) {
        super(context, attrs);

        init(getContext());
    }

    public AccountDetailsView(@NonNull Context context,
                              @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs,
                              int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        init(getContext());
    }

    public AccountDetailsView(@NonNull Context context,
                              @Nullable @org.jetbrains.annotations.Nullable AttributeSet attrs,
                              int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);

        init(getContext());
    }

    public void init(Context context) {
        inflate(context, R.layout.layout_account_details_2, this);

        FirebaseUser user = FirestoreUser.getCurrentFirebaseUser();

        try {
            setUsernameInitials(user);
        } catch (Exception e) {
            e.printStackTrace();
        }
        setContainerVisibility(user);

    }

    private void setContainerVisibility(@Nullable FirebaseUser user) {
        if(user == null) {
            setVisibility(GONE);
        } else {
            setVisibility(VISIBLE);
        }
    }

    private void setUsernameInitials(@Nullable FirebaseUser user)
            throws Exception {

        if(user == null) {
            throw new FirestoreUser.NullFirebaseUserException();
        }

        AccountIconView accountIcon = findViewById(R.id.account_icon);
        if(accountIcon != null) {
            accountIcon.createAccountInitials(user.getDisplayName());
        }
    }

}

File: app-view/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/views/account/AccountDetailsView.kt
package com.tritiumgaming.phasmophobiaevidencepicker.views.account

import android.content.Context
import android.util.AttributeSet
import androidx.constraintlayout.widget.ConstraintLayout
import com.tritiumgaming.phasmophobiaevidencepicker.R
import com.tritiumgaming.phasmophobiaevidencepicker.firebase.firestore.transactions.user.FirestoreUser
import com.google.firebase.auth.FirebaseUser

class AccountDetailsView : ConstraintLayout {
    constructor(context: Context) : super(context) {
        init(getContext())
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(getContext())
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
            super(context, attrs, defStyleAttr) {
        init(getContext())
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
            super(context, attrs, defStyleAttr, defStyleRes) {
        init(getContext())
    }

    fun init(context: Context?) {
        inflate(context, R.layout.layout_account_details_2, this)

        val user = FirestoreUser.currentFirebaseUser
        setUsernameInitials(user)

        setContainerVisibility(user)
    }

    private fun setContainerVisibility(user: FirebaseUser?) {
        visibility = if (user == null) { GONE } else { VISIBLE }
    }

    @Throws(Exception::class)
    private fun setUsernameInitials(user: FirebaseUser?) {
        val accountIcon = findViewById<AccountIconView>(R.id.account_icon)
        accountIcon?.createAccountInitials(user?.displayName)
    }
}
